openapi/paths.libsonnet

```diff
@@ -0,0 +1,110 @@
+// Copyright © 2018 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+local oapi = import 'openapi-jsonnet/v3.0.0/openapi.libsonnet';
+local ops = import 'operations.libsonnet';
+local params = import 'parameters.libsonnet';
+local pathItem = oapi.pathItem;
+
+local endpoints = {
+  oci:: {
+    base:: '/v2',
+    catalog:: '/v2/_catalog',
+    tags:: '/v2/{namespace}/{repository}/tags/list',
+    manifest:: '/v2/{namespace}/{repository}/manifests/{reference}',
+    blob:: '/v2/{namespace}/{repository}/blobs/{digest}',
+    upload:: '/v2/{namespace}/{repository}/blobs/uploads',
+    uploadId:: '/v2/{namespace}/{repository}/blobs/uploads/{uuid}',
+  },
+};
+
+local items = {
+  oci:: {
+    local base =
+      pathItem.new()
+      .addOperation('GET', ops.oci.verify),
+
+    local catalog =
+      pathItem.new()
+      .addParameter(params.n)
```

Comment:
Why string? Why required?

Reply:
There isn't an official jsonschema for catalog. It's an optional endpoint so I'm not too worried about it.

Reply:
The specification must be accurate and precise. I believe that this parameter should be optional. Is there any docs mention ```n``` as required? 

The JSONSchema specification can be used for generated API clients, and then the optional parameter makes huge difference for compilator, code, library usage etc.

Reply:
Documentation mention:
```
The catalog for a given registry can be retrieved with the following request:

GET /v2/_catalog
```

It mean there is no required parameter like ```n``` or ```limits```.

Reply:
Sounds good. I'll work on this one as well.

Reply:
Fixed.